### DISCUSSION TITLE
useCookie how to delete cookie?

### DISCUSSION THREAD

**Discussion by vedmant** [nuxt/nuxt#16300] [Questions]
I'm looking into documentation for [useCookie](https://nuxt.com/docs/api/composables/use-cookie#usecookie) but there is no example how I can delete cookie.

**Comment by Diizzayy** [comment]
@vedmant A cookie can be deleted as seen below

```ts
const myCookie = useCookie('<cookie_name>', { /* cookie_options */ })

// set cookie
myCookie.value = '<cookie_value>'

// delete cookie
myCookie.value = null
```

**Comment by vedmant** [comment]
@Diizzayy Btw, when I just need to delete cookie, do I need to specify all the options again that I used to set cookie? like:
```ts
const cookie = useCookie('sso-auth', { path: '/', maxAge: 60 * 60 * 24 * 30 })
cookie.value = null
```

**Reply by Diizzayy** [reply]
> do I need to specify all the options again that I used to set cookie

@vedmant not at all, just specifying the cookie name should be suffice if I'm not mistaken.

**Comment by acidjazz** [comment]
null and undefined are not deleteing my cookie

```ts
    console.log('setting this.token.value to null..')
    this.token.value = null
    console.log(this.token.value)
    if (process.client) console.log(document.cookie)
```

<img width="490" alt="image" src="https://user-images.githubusercontent.com/967369/214750199-39954c5b-487a-402e-8a05-8c51b5bdf25c.png">

**Reply by vedmant** [reply]
@acidjazz Where do you use `useCookie()` in your code?

**Reply by acidjazz** [reply]
here it is at the top of the class:
```ts
  public token = useCookie<string | undefined>('token', { path: '/', maxAge: 60 * 60 * 24 * 30 })
```

adding this vanilla JS properly removed the cookie:
```ts
    if (process.client) document.cookie = 'token=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/;'
```
